tutorials/kube-mqtt/index.md

```diff
@@ -0,0 +1,267 @@
+1.  [Enable billing for your project](https://cloud.google.com/billing/docs/how-to/modify-project).
+1.  Use [Cloud Shell][shell] or install the [Google Cloud SDK][sdk].
+
+
+In this tutorial, you may open multiple shell tabs or sessions. You should set these environment variables in each one:
+
+	# the project should already be set in cloud shell
+	gcloud config set project <project-id>
+	
+	export PROJECT=$(gcloud config list project --format "value(core.project)" )
+	export REGION=us-central1
+	export REGISTRY=gadgets
+	export DEVICE=bridge
+	
+	
+[console]: https://console.cloud.google.com/
+[shell]: https://cloud.google.com/shell/
+[sdk]: https://cloud.google.com/sdk/
+
+Enable APIs:
+	
+	gcloud services enable \
+	cloudiot.googleapis.com \
+	container.googleapis.com \
+	containerregistry.googleapis.com \
+	pubsub.googleapis.com \
+	cloudbuild.googleapis.com
+		
+	
+## Build the bridge manager container
+
+For Mosquitto you will use the default published container from Docker. But you need another container in the deployment which manages the credentials that allow the bridge to connect to Google Cloud.
+
+	cd refresher-container
+	gcloud builds submit --tag gcr.io/$PROJECT/refresher .
+
+## Create Cloud resources:
+
+### Create the device keys
+
+	cd ../bridge
+	openssl ecparam -genkey -name prime256v1 -noout -out ec_private.pem
+	openssl ec -in ec_private.pem -pubout -out ec_public.pem
+
+### Create IoT Core resources
+
+	gcloud pubsub topics create device-events
+	gcloud pubsub subscriptions create debug --topic device-events
+	
+	gcloud iot registries create $REGISTRY \
+	--region=$REGION \
+	--event-notification-config topic=device-events
+	
+	gcloud iot devices create $DEVICE \
+	--region $REGION \
+	--registry $REGISTRY \
+	--public-key path=ec_public.pem,type=es256-pem
+
+This creates an IoT Core device called "bridge" that you will use to authenticate the bridge to IoT Core with.
+
+The pubsub topics are used to verify that data is being bridged to cloud correctly.
+
+## Deploy the bridge to Kubernetes
+
+### Update project specific settings
+
+Get the specific address for the bridge manager container:
+
+	gcloud container images describe gcr.io/$PROJECT/refresher --format="value(image_summary.fully_qualified_digest)"
+
+In the repo file `bridge/project-image.yaml`, add the output from the previous command as `image` .
+
+If running the Kubernetes cluster outside the Cloud Project, make the images public:
+
+	gsutil iam ch allUsers:objectViewer gs://artifacts.$PROJECT.appspot.com
+
+Note this will make the container registry for the project public.
+
+In production environments, you would provision the cluster with the required [image pull secrets](http://docs.heptio.com/content/private-registries/pr-gcr.html).
+
+Edit PROJECT_ID value in `device-config.yaml`
+
+### Deploy
+
+You are going to use the built in version of [kustomize](https://kustomize.io/) in recent kubectl versions to deploy the solution to kubernetes.
```

Comment:
You are going to use the built-in version of [kustomize](https://kustomize.io/) in recent `kubectl` versions to deploy the solution to Kubernetes.